### PR TITLE
Adding `logfire` examples to `integrations` page

### PR DESCRIPTION
**Pull request by sydney-runkle**
Got a start here on `logfire` examples. We'll probably want to add some more info related to tracing here - looking for feedback on what context we want to use for that demo.

### CODE REVIEW

**Review by samuelcolvin — COMMENTED** [review]
I'll read in more detail tomorrow, but in general this looks great.

@dmontagu can you review too?

File: docs/img/basic_logfire.png
Comment (samuelcolvin) [inline docs/img/basic_logfire.png:1]: looks like you need to run these through tinify - https://gist.github.com/samuelcolvin/349a845497be02b8c69d1d90d7f91ac3

File: docs/integrations/logfire.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+Pydantic integrates seamlessly with **Logfire**, an observability platform built on the same belief as our open source library — that the most powerful tools can be easy to use.
```
Comment (samuelcolvin) [inline docs/integrations/logfire.md:1]:
```suggestion
Pydantic integrates seamlessly with **Pydantic Logfire**, an observability platform built by us on the same belief as our open source library — that the most powerful tools can be easy to use.
```

File: mkdocs.yml [hunk 2]
```diff
@@ -168,6 +168,7 @@ nav:
     - Linting: integrations/linting.md
   - Production Tools:
     - AWS Lambda: integrations/aws_lambda.md
+    - Pydantic Logfire: integrations/logfire.md
```
Comment (samuelcolvin) [inline mkdocs.yml:171]: I think we can come above AWS!

I would say we should alos put this section before "Dev Tools`, and be the default on this tab.

File: docs/img/basic_logfire.png
Comment (samuelcolvin) [inline docs/img/basic_logfire.png:1]: this looks wrong, details of the user are now shown in the message on the left.

File: docs/integrations/logfire.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+1. The `logfire.configure()` call is all you need to instrument your project with Logfire.
+2. The `logfire.info()` call logs the `user` object to Logfire, with builtin support for Pydantic models.
+
+![basic pydantic logfire usage](../img/basic_logfire.png)
+
+### Pydantic Instrumentation
+
+You can even record information about the validation process automatically by
+using the builtin Pydantic integration:
```
Comment (samuelcolvin) [inline docs/integrations/logfire.md:45]: we should link to the logfire docs here.

**Review by samuelcolvin — COMMENTED** [review]
Otherwise LGTM.

File: docs/img/basic_logfire.png
Comment (sydney-runkle) [reply docs/img/basic_logfire.png:1]: Hmm, I took these screenshots yesterday?

Comment (sydney-runkle) [reply docs/img/basic_logfire.png:1]: Done, was a screen width thing.

**Pull request merged**